### PR TITLE
RavenDB-14080 - Backup is encrypted regardless explicitly disabling it - cannot restore such backup

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-14080

### CODE REVIEW

**Review by ml054 — APPROVED** [review]
Studio part looks good.

File: src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts [hunk 1]
```diff
@@ -188,16 +188,8 @@ class encryptionSettings {
...
-        {
-            return {
-                EncryptionMode: "None",
-                Key: null
-            };
-        }
-        
         return {
-            EncryptionMode: this.mode(),
+            EncryptionMode: !this.enabled() ? "None" : this.mode(),
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: can we reverse logic ?
this.enabled() ? this.mode()  : "None"

Comment (grisha-kotler) [reply src/Raven.Studio/typescript/models/database/tasks/periodicBackup/encryptionSettings.ts]: done

**Review by ayende — APPROVED** [review]

File: src/Raven.Server/Documents/PeriodicBackup/BackupTask.cs [hunk 2]
```diff
@@ -737,29 +736,17 @@ private void AddInfo(string message)
...
-                return new EncryptingXChaCha20Poly1305Stream(fileStream,
-                    _database.MasterKey);
+            if (_database.MasterKey != null && _configuration.BackupEncryptionSettings == null)
+                return new EncryptingXChaCha20Poly1305Stream(fileStream, _database.MasterKey);
 
-            if (_configuration?.BackupEncryptionSettings?.EncryptionMode == EncryptionMode.UseDatabaseKey)
-                return new EncryptingXChaCha20Poly1305Stream(fileStream,
-                    _database.MasterKey);
+            if (_configuration.BackupEncryptionSettings.EncryptionMode == EncryptionMode.UseDatabaseKey)
+                return new EncryptingXChaCha20Poly1305Stream(fileStream, _database.MasterKey);
```
Comment (ppekrol) [inline src/Raven.Server/Documents/PeriodicBackup/BackupTask.cs:746]: if `_database.MasterKey` is null we need to throw?

Comment (grisha-kotler) [reply src/Raven.Server/Documents/PeriodicBackup/BackupTask.cs:746]: It's validated before the backup starts:
https://github.com/grisha-kotler/ravendb/blob/bfeb644d706ca5e844c8f75dc474298a2d99f0c5/src/Raven.Server/Documents/PeriodicBackup/BackupTask.cs#L645

**Pull request merged**